### PR TITLE
feat(autorole): add level syntax support

### PR DESCRIPTION
**Pull request by kyranet**

### CODE REVIEW

File: src/languages/en-US/commands/social.json [hunk 1]
```diff
@@ -1,9 +1,13 @@
 {
 	"autoRoleUpdateConfigured": "This role is already configured as an autorole. Use the remove type instead.",
 	"autoRoleUpdateUnconfigured": "This role is not configured as an autorole. Use the add type instead.",
-	"autoRoleUpdate": "Updated autorole: {{role.name}} ({{role.id}}). Points required: {{points}} (before: {{before}})",
-	"autoRoleRemove": "Removed the autorole: {{role.name}} ({{role.id}}), which required {{before}} points.",
-	"autoRoleAdd": "Added new autorole: {{role.name}} ({{role.id}}). Points required: {{points}}",
+	"autoRoleInvalidLevel": "I could not resolve `{{parameter}}` into a level, you must input either a number (for example: `1500` or `51000`) or a number suffixed with `l` or `L` (for example: `10L`, `20l`)",
```
Comment (favna) [inline src/languages/en-US/commands/social.json]:
```suggestion
	"autoRoleInvalidLevel": "I could not resolve `{{parameter}}` into a level, you must input either a number (for example: `1500` or `51000`) or a number suffixed with `l` or `L` (for example: `10L` or `20l`)",
```

File: src/languages/en-US/commands/social.json [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
+	"autoRoleInvalidNegativeOrZeroLevel": "I could not resolve `{{parameter}}` into a valid level, they cannot be zero nor negative.",
```
Comment (favna) [inline src/languages/en-US/commands/social.json]:
```suggestion
	"autoRoleInvalidNegativeOrZeroLevel": "I could not resolve `{{parameter}}` into a valid level. A level cannot be zero or lower.",
```

File: src/languages/en-US/commands/social.json [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
+	"autoRoleTooSmall": "I could not resolve `{{parameter}}` into a valid level, it resolves to {{points, number}} points, which is below {{minimum, number}}.",
+	"autoRoleTooBig": "I could not resolve `{{parameter}}` into a valid level, it resolves to {{points, number}} points, which is above {{maximum, number}}.",
```
Comment (favna) [inline src/languages/en-US/commands/social.json]:
```suggestion
	"autoRoleTooLow": "I could not resolve `{{parameter}}` into a valid level, it resolves to {{points, number}} points, which is below {{minimum, number}}.",
	"autoRoleTooHigh": "I could not resolve `{{parameter}}` into a valid level, it resolves to {{points, number}} points, which is above {{maximum, number}}.",
```

big and small are not how you describe minimum and maximum when it comes to numbers. Big and small are used to describe groups of things, or the length of an object or animal/person.
Update language keys accordingly.

File: src/lib/i18n/languageKeys/keys/commands/Levelling.ts [hunk 4]
```diff
@@ -24,6 +24,10 @@ export const AutoRoleRemove = FT<{ role: Role; before: number }, string>('comman
 export const AutoRoleUpdate = FT<{ role: Role; points: number; before: number }, string>('commands/social:autoRoleUpdate');
 export const AutoRoleUpdateConfigured = T<string>('commands/social:autoRoleUpdateConfigured');
 export const AutoRoleUpdateUnconfigured = T<string>('commands/social:autoRoleUpdateUnconfigured');
+export const AutoRoleInvalidLevel = FT<{ parameter: string }, string>('commands/social:autoRoleInvalidLevel');
+export const AutoRoleInvalidNegativeOrZeroLevel = FT<{ parameter: string }, string>('commands/social:autoRoleInvalidNegativeOrZeroLevel');
+export const AutoRoleTooSmall = FT<{ parameter: string; minimum: number; maximum: number }, string>('commands/social:autoRoleTooSmall');
+export const AutoRoleTooBig = FT<{ parameter: string; minimum: number; maximum: number }, string>('commands/social:autoRoleTooBig');
```
Comment (favna) [inline src/lib/i18n/languageKeys/keys/commands/Levelling.ts]:
```suggestion
export const AutoRoleTooLow = FT<{ parameter: string; minimum: number; maximum: number }, string>('commands/social:autoRoleTooLow');
export const AutoRoleTooHigh = FT<{ parameter: string; minimum: number; maximum: number }, string>('commands/social:autoRoleTooHigh');
```

File: src/languages/en-US/commands/social.json [hunk 5]
```diff
@@ -1,9 +1,13 @@
...
+	"autoRoleUpdate": "Updated autorole: {{role.name}} (`{{role.id}}`). Points required: {{points, number}} (before: {{before, number}})",
+	"autoRoleRemove": "Removed the autorole: {{role.name}} (`{{role.id}}`), which required {{before, number}} points.",
+	"autoRoleAdd": "Added new autorole: {{role.name}} (`{{role.id}}`). Points required: {{points, number}}",
```
Comment (favna) [inline src/languages/en-US/commands/social.json]:
```suggestion
	"autoRoleUpdate": "Updated autorole: {{role}}. Points required: {{points, number}} (before: {{before, number}})",
	"autoRoleRemove": "Removed the autorole: {{role}}, which required {{before, number}} points.",
	"autoRoleAdd": "Added new autorole: {{role}}. Points required: {{points, number}}",
```

Pass `role.toString()` into the parameter and set `allowedMentions` to not ping the role.

File: src/commands/Levelling/autorole.ts [hunk 6]
```diff
@@ -104,4 +105,55 @@ export class UserCommand extends SkyraCommand {
...
+				identifier: LanguageKeys.Commands.Social.AutoRoleTooSmall,
+				context: { minimum: 1, maximum: this.maximumPoints, points }
+			});
+		}
+
+		if (points > this.maximumPoints) {
+			return Args.error({
+				argument,
+				parameter,
+				identifier: LanguageKeys.Commands.Social.AutoRoleTooBig,
```
Comment (favna) [inline src/commands/Levelling/autorole.ts]:
```suggestion
				identifier: LanguageKeys.Commands.Social.AutoRoleTooHigh,
```

File: src/commands/Levelling/autorole.ts [hunk 7]
```diff
@@ -104,4 +105,55 @@ export class UserCommand extends SkyraCommand {
...
+		if (level <= 0) {
+			return Args.error({ argument, parameter, identifier: LanguageKeys.Commands.Social.AutoRoleInvalidNegativeOrZeroLevel });
+		}
+
+		const points = Math.floor((level / 0.2) ** 2);
+		if (points < 1) {
+			return Args.error({
+				argument,
+				parameter,
+				identifier: LanguageKeys.Commands.Social.AutoRoleTooSmall,
```
Comment (favna) [inline src/commands/Levelling/autorole.ts]:
```suggestion
				identifier: LanguageKeys.Commands.Social.AutoRoleTooLow,
```

**Review by favna — CHANGES_REQUESTED** [review]
```ts
export const AutoRoleAdd = FT<{ role: string; points: number }, string>('commands/social:autoRoleAdd');
export const AutoRoleRemove = FT<{ role: string; before: number }, string>('commands/social:autoRoleRemove');
export const AutoRoleUpdate = FT<{ role: string; points: number; before: number }, string>('commands/social:autoRoleUpdate');
```

For the remarks concerning mentioning the role with `allowedMentions` set to not ping it.

**Review by vladfrangu — APPROVED** [review]
Honestly I'd do `L<number>` for level argument but w/e, LGTM

**Pull request merged**